### ISSUE TITLE
Point Cloud Support

### ISSUE THREAD

**Issue by ilkedemir** [mikedh/trimesh#137]
In trimesh version 2.22.27, I was able to create a vertex-only model. Then I was able to process it to downsample the points. In version 2.26.0, I get the following error. Can you provide the backward compatibility for vertex-only models? 

Example:
```
    v=[np.array((a,a+1,a+2)) for a in range(0,100)] # or any other set of points
    p=trimesh.Trimesh(vertices=v)
```

> Error:
> Traceback (most recent call last):
> 
>   File "C:/Users/idemir/Downloads/try_mayavi.py", line 753, in <module>
>     p=trimesh.Trimesh(vertices=v)
> 
>   File "C:\Users\idemir\anaconda\envs\oldqt\lib\site-packages\trimesh\base.py", line 175, in __init__
>     self.process()
> 
>   File "C:\Users\idemir\anaconda\envs\oldqt\lib\site-packages\trimesh\base.py", line 205, in process
>     self.remove_infinite_values()
> 
>   File "C:\Users\idemir\anaconda\envs\oldqt\lib\site-packages\trimesh\base.py", line 991, in remove_infinite_values
>     faces = np.isfinite(self.faces).all(axis=1)
> 
>   File "C:\Users\idemir\anaconda\envs\oldqt\lib\site-packages\numpy\core\_methods.py", line 41, in _all
>     return umr_all(a, axis, dtype, out, keepdims)
> 
> ValueError: 'axis' entry is out of bounds

Related line of code:
https://github.com/mikedh/trimesh/blob/b5e80702f6de4f62035bf8b405b68bab0e5f8d8a/trimesh/base.py#L991

Digging a little bit more, the error emerges from the change from former line 163 to current line 174 of base.py. Previous check:

```
    if (process and
        (vertices is not None) and
            (faces is not None)):
```

Current check:
```
    if process or validate:
```

So my generation of vertex-only models was passing the check as `faces` was `None`. Now it does not pass and `process` is called, so the error above occurs. I would appreciate if this bug (?) is resolved, so that I can roll to the latest. Thanks!

**Comment by mikedh** [comment]
Hey, that's an interesting one. I just added more checking for empty data just now which should fix your issue. I do think however you'd be better off just using the `unique_rows` functionality to do the same thing on straight numpy arrays without using a mesh object:

```
    v = np.random.random((1000,3))
    v[np.floor(np.random.random(100)*len(v)).astype(int)] = v[0]

    unique, inverse = trimesh.grouping.unique_rows(v)
    v = v[unique]
```

Or, you could use the PointCloud object (which as a side benefit can be visualized with a `show()` call:
```
p = trimesh.points.PointCloud(v)
p.show()
```

**Comment by ilkedemir** [comment]
Thanks for the quick fix, I confirm that version 2.26.9 does not break.
